llvm-bazel/llvm-project-overlay/clang/BUILD

```diff
@@ -0,0 +1,1976 @@
+        ":driver",
+        ":frontend",
+        ":index",
+        ":lex",
+        ":rewrite",
+        ":sema",
+        ":tooling",
+        "//llvm:BitstreamReader",
+        "//llvm:FrontendOpenMP",
+        "//llvm:Support",
+        "//llvm:config",
+    ],
+)
+
+cc_library(
+    name = "c-bindings",
+    hdrs = glob(["include/clang-c/*.h"]),
+    deps = [
+        ":libclang",
+        #"//clang-tools-extra/clang-include-fixer:include-fixer-plugin",
+    ],
+    alwayslink = 1,
+)
+
+cc_binary(
+    name = "libclang.so",
+    linkopts = [
+        "-nodefaultlibs",
+        "-lc_nonshared",
+        "-lgcc",
+        "-lgcc_eh",
+        "-Wno-unused-command-line-argument",
+    ],
+    linkshared = 1,
+    deps = [":c-bindings"],
+)
+
+filegroup(
+    name = "python-sources",
+    srcs = [
+        "bindings/python/clang/cindex.py",
+        "bindings/python/clang/enumerations.py",
+    ],
+)
+
+filegroup(
+    name = "python-cindex-examples",
+    srcs = [
+        "bindings/python/examples/cindex/cindex-dump.py",
+        "bindings/python/examples/cindex/cindex-includes.py",
+    ],
+)
+
+cc_binary(
+    name = "c-index-test",
+    testonly = 1,
+    srcs = [
+        "tools/c-index-test/c-index-test.c",
+        "tools/c-index-test/core_main.cpp",
+    ],
+    copts = [
+        "-Wno-uninitialized",
+    ],
+    stamp = 0,
+    deps = [
+        ":ast",
+        ":basic",
+        ":c-bindings",
+        ":codegen",
+        ":config",
+        ":frontend",
+        ":index",
+        ":lex",
+        ":parse",
+        ":sema",
+        ":serialization",
+        "//llvm:Core",
+        "//llvm:MC",
+        "//llvm:Support",
+    ],
+)
+
+cc_binary(
+    name = "arcmt-test",
+    testonly = 1,
+    srcs = ["tools/arcmt-test/arcmt-test.cpp"],
+    stamp = 0,
+    deps = [
+        ":arc_migrate",
+        ":ast",
+        ":basic",
+        ":frontend",
+        ":frontend_rewrite",
+        ":lex",
+        "//llvm:Support",
+    ],
+)
+
+cc_binary(
+    name = "c-arcmt-test",
+    testonly = 1,
+    srcs = ["tools/c-arcmt-test/c-arcmt-test.c"],
+    copts = ["-std=gnu99"],
+    stamp = 0,
+    deps = [
+        ":c-bindings",
+        ":codegen",
+        "//llvm:MC",
+        "//llvm:Support",
+    ],
+)
+
+cc_binary(
+    name = "clang-import-test",
+    testonly = 1,
+    srcs = glob([
+        "tools/clang-import-test/*.cpp",
+        "tools/clang-import-test/*.h",
+    ]),
+    stamp = 0,
+    deps = [
+        ":ast",
+        ":basic",
+        ":codegen",
+        ":driver",
+        ":frontend",
+        ":lex",
+        ":parse",
+        "//llvm:Core",
+        "//llvm:Support",
+    ],
+)
+
+cc_library(
+    name = "clang-driver",
+    srcs = glob([
+        "tools/driver/*.cpp",
+        "tools/driver/*.h",
+    ]),
+    copts = [
+        # Disable stack frame size checks in the driver because
+        # clang::ensureStackAddressSpace allocates a large array on the stack.
+        "$(STACK_FRAME_UNLIMITED)",
```

Comment:
We use this basically everywhere else without comment. Is this usage significantly different or more interesting?

Reply:
It is a bit more interesting. This code has a very specific bit of code that intentionally uses a large amount of stack and would trigger even a really reliable stack frame limit warning. Most of the other cases are somewhat incidental and maybe even debatable. But this one is really recognizing a very unusual bit of code for manipulating the stack in a weird way to cope with Linux kernel bugs. (See the comments above the mentioned function for all the wild details.)